packages/terra-data-grid/src/subcomponents/ColumnResizeHandle.jsx

```diff
@@ -85,28 +119,109 @@ const ColumnResizeHandle = (props) => {
+      case KeyCode.KEY_SPACE:
+      case KeyCode.KEY_RETURN:
+        setNavigationEnabled(false);
+        // Lock focus into component
+        resizeHandleRef.current.focus();
+
+        // Assistive technologies should avoid announcing aria-label while focus locked, but announce aria-valueText instead
+        setIsAriaLabel(false);
+        setIsAriaValueText(true);
+        // Assistive technologies should make announcement for focus locked
+        columnContext.setColumnHeaderAriaLiveMessage(intl.formatMessage({ id: 'Terra.worklist-data-grid.cell-focus-trapped' }));
+        event.stopPropagation();
+        event.preventDefault();
+        break;
+      case KeyCode.KEY_ESCAPE:
+        // Assistive technologies should stop announcing aria-valueText once focus unlocked
+        setIsAriaValueText(false);
+        // Release focus lock
+        columnContext.setColumnHeaderAriaLiveMessage(intl.formatMessage({ id: 'Terra.worklist-data-grid.resume-navigation' }));
+        setNavigationEnabled(true);
+        break;
+      case KeyCode.KEY_RIGHT:
+        if (onResizeHandleChange && !isNavigationEnabled) {
+          // Increase column width
+          onResizeHandleChange(columnIndex, columnResizeIncrement);
+        }
+        break;
+      case KeyCode.KEY_LEFT:
+        if (onResizeHandleChange && !isNavigationEnabled) {
+          // Decrease column width
+          onResizeHandleChange(columnIndex, -columnResizeIncrement);
+        }
+        break;
+      default:
+    }
+
+    if (!isNavigationEnabled) {
+      event.stopPropagation();
+      event.preventDefault();
+    }
+  };
+
+  // Prevent click event propagation
+  const onClick = (event) => {
+    // Prevent event bubbling since necessary actions are handled by this component
+    event.stopPropagation();
+  };
+
+  const onBlur = () => {
+    setNavigationEnabled(true);
+    setIsAriaLabel(false);
+    setIsActive(false);
+  };
+
   return (
-    /* eslint-disable react/forbid-dom-props */
-    // eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/interactive-supports-focus
+    // eslint-disable-next-line jsx-a11y/no-noninteractive-element-interactions, jsx-a11y/no-static-element-interactions
     <div
       ref={resizeHandleRef}
       draggable
-      role="slider"
+      role={isNavigationEnabled ? null : 'slider'}
       tabIndex={-1}
       aria-hidden={!isActive}
-      aria-valuemin={minimumWidth}
-      aria-valuenow={columnWidth}
-      aria-valuemax={maximumWidth}
-      aria-label={columnText}
-      aria-valuetext={intl.formatMessage({ id: 'Terra.dataGrid.resizeHandleValueText' }, { columnWidth })}
-      style={{ height: `${height}px` }}
+      aria-valuemin={isActive ? minimumWidth : null}
+      aria-valuenow={isActive ? columnWidth : null}
+      aria-valuemax={isActive ? maximumWidth : null}
+      aria-label={isAriaLabel ? intl.formatMessage({ id: 'Terra.worklist-data-grid.resize-handle-template' }, { columnText }) : null}
```

Comment:
For some reason this 'Terra.worklist-data-grid.resize-handle-template' is being set as the actual aria-label upon focusing the resize handle.

Reply:
This got magically fixed on its own.